app/validators/json_validator.rb

```diff
@@ -0,0 +1,22 @@
+class JsonValidator
+  def initialize(schema_name, payload)
+    @schema_dir = Rails.root.join("public/schemas")
+    @schema_name = load_schema(schema_name)
```

Comment:
for clarity i would have a @schema_name and a @schema var. The @schema being the loaded :"schema"  of name @schema_name 

```
class JsonValidator
   def initialize(schema_name, payload)
     @schema_name = schema_name
     @schema = load_schema
   end
   
   
   private
   
   attr_reader :schema_name, :schema
   
   ...can amend to use attr_reader fpor schema_name and/or schema elsewhere then
```

Do you want to pair on this?!